### ISSUE TITLE
review release tracks and versioning scheme

### ISSUE THREAD

**Issue by q9f** [ChainSafe/lodestar#2990]
follow-up after #2950 - continuing discussion from #2958 

> > > @q9f Another better option would be to use nightly versioning as `0.27.1-dev.a6b280` instead of what we have right now that's just a counter `0.27.1-dev.31` +1
> > 
> > 
> > I just [checked](https://github.com/ChainSafe/lodestar/runs/3354937219), the nighly release version is `0.28.2-dev.3+022659b5fc0` according to Github actions (appending the `--canary` git sha option) but the NPM registry just truncates everything after the `+` to strict (?) semver and it just [remains](https://www.npmjs.com/package/@chainsafe/lodestar/v/0.28.2-dev.2) `0.28.2-dev.3`. I found this: [lerna/lerna#1893](https://github.com/lerna/lerna/issues/1893)
> > > Association with the current NPM versions is very hard. If the published version is `foo@0.1.5-alpha.124`, I have to count 124 commits up from `0.1.4` or go through CI logs to figure out which commit that version is tied to. If it were `foo@0.1.5-alpha.124.3d6bee55` or `foo@0.1.5-alpha.3d6bee55` it would be much easier.
> > > It looks like Lerna is publishing with the `+${sha}` suffix, but it seems NPM is stripping it out. If that's the case, why is this format [not configurable](https://github.com/lerna/lerna/blob/master/commands/publish/index.js#L359)?
> > 
> > 
> > At first glance, there seems no easy way to force lerna to do that properly.
> 
> What if we write our own version format? Not use any counter at all and only the commit hash:
> 
> ```
> 0.28.2-dev.022659b5fc0
> ```

@dapplion I'm totally down to change the version strings to our custom format. This, however, might change the release process (and ci scripts) that basically just used `lerna publish`.

I'll leave this open for discussion and maybe I can find a suitable alternative to `lerna` during the week.

**Comment by dapplion** [comment]
@q9f Doesn't lerna support providing a custom version string?

**Comment by q9f** [comment]
Lerna just allows strict semver. We could hack the sha into the preid but I'm wondering if there are alternatives.

no custom versions: https://github.com/lerna/lerna/issues/2813

**Comment by dapplion** [comment]
> Lerna just allows strict semver. We could hack the sha into the preid but I'm wondering if there are alternatives.
> 
> no custom versions: [lerna/lerna#2813](https://github.com/lerna/lerna/issues/2813)

Or run a search an replace? hahaha

**Comment by q9f** [comment]
Closing in favor of #3036
